app/models/location_aware_booking_requests.rb

```diff
@@ -0,0 +1,21 @@
+class LocationAwareBookingRequests < SimpleDelegator
+  def initialize(booking_requests, booking_location)
+    @booking_location = booking_location
+    super(booking_requests)
+  end
+
+  def each
+    return enum_for(:each) unless block_given?
+
+    __getobj__.each do |booking_request|
+      yield LocationAwareBookingRequest.new(
+        booking_location: booking_location,
+        booking_request: booking_request
+      )
+    end
```

Comment:
Because it needs to answer to `#empty?` and other enumerable behaviour - is the short explanation. I could just delegate `#empty?` and `#each` without deriving from `SimpleDelegator` sure, but that's just another approach with the same result and I'm not sure I see the benefits either way.

What do you think about this:

```ruby
class LocationAwareBookingRequests
  def initialize(booking_requests, booking_location)
     # blah
  end

  def all
    @all ||= @booking_requests.map { |b| LocationAwareBookingRequest.new(b, @location)
  end
end
```

Changing the interface slightly but achieving the same. Albeit a little simpler.